### PR TITLE
[Components with more than 1 service] LPS-189681 Refactor HttpMethodScopeLogic component

### PR DESCRIPTION
**Pull request by Valmir-unicap**
Related: https://liferay.atlassian.net/browse/LPS-189681

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:startup

**Comment by Valmir-unicap** [comment]
ci:test:security

**Comment by Valmir-unicap** [comment]
ci:stop

**Comment by Valmir-unicap** [comment]
ci:help

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:startup

**Comment by Valmir-unicap** [comment]
ci:test:security

**Comment by Kvale1733** [comment]
ci:test:oauth2

**Comment by Kvale1733** [comment]
ci:stop

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/scope/logic/HttpMethodScopeLogic.java [hunk 1]
```diff
@@ -49,58 +40,22 @@ public boolean check(
...
-
 	@Activate
 	protected void activate(
 		BundleContext bundleContext, Map<String, Object> properties) {
 
 		_bundleContext = bundleContext;
 
-		_ignoreMissingScopes = new HashSet<>(
-			StringPlus.asList(properties.get("ignore.missing.scopes")));
+		_ignoreMissingScopes = new HashSet<>();
```
Comment (rafaprax) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/scope/logic/HttpMethodScopeLogic.java:54]: Hey @Valmir-unicap, you do not need this entire method and the variables within it. Please remove.

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/scope/util/HttpMethodScopeLogicUtil.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		try {
+			String applicationName = GetterUtil.getString(
+				propertyAccessorFunction.apply("osgi.jaxrs.name"));
+
+			Object ignoreMissingScopesObject = propertyAccessorFunction.apply(
+				"ignore.missing.scopes");
+
+			Set<String> ignoreMissingScopes = _defaultIgnoreMissingScopes;
+
+			Bundle bundle = FrameworkUtil.getBundle(
```
Comment (rafaprax) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/scope/util/HttpMethodScopeLogicUtil.java:45]: initialize the bunblecContext variable into a static block

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:test:oauth2

**Comment by Valmir-unicap** [comment]
ci:test:startup

**Comment by Valmir-unicap** [comment]
ci:test:security

**Comment by Kvale1733** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:test:security

**Comment by Kvale1733** [comment]
ci:test:oauth2

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by catarineaguiar** [comment]
ci:stop

**Comment by Valmir-unicap** [comment]
ci:test:oauth2

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/scope/util/HttpMethodScopeLogicUtil.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
+		HttpMethodScopeLogicUtil.class);
+
+	private static final BundleContext _bundleContext;
+	private static final Set<String> _defaultIgnoreMissingScopes =
+		new HashSet<String>() {
+			{
+				_defaultIgnoreMissingScopes.add("HEAD");
```
Comment (rafaprax) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/scope/util/HttpMethodScopeLogicUtil.java:94]: Hey @Valmir-unicap, you do not need to use the `_defaultIgnoreMissingScopes` reference here, just call the `add` method directly.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
Hey @Valmir-unicap, Please resend when you apply the changes requested. Thx

**Pull request closed without merging**